RecoCTPPS/TotemRPLocal/interface/CTPPSDiamondTrackRecognition.h

```diff
@@ -4,76 +4,110 @@
-#include "TF1.h"
 
 /**
- * \brief Class performing smart reconstruction for CTPPS Diamond Detectors.
+ * \brief Class performing smart reconstruction for PPS Diamond Detectors.
  * \date Jan 2017
 **/
-class CTPPSDiamondTrackRecognition
+class CTPPSDiamondTrackRecognition : public CTPPSTimingTrackRecognition<CTPPSDiamondLocalTrack, CTPPSDiamondRecHit>
 {
   public:
-    CTPPSDiamondTrackRecognition( const edm::ParameterSet& );
-    ~CTPPSDiamondTrackRecognition();
-
-    /// Reset the list of hits
-    void clear();
+    CTPPSDiamondTrackRecognition( const edm::ParameterSet& iConfig ) :
+      CTPPSTimingTrackRecognition<CTPPSDiamondLocalTrack, CTPPSDiamondRecHit>( iConfig ) {}
 
+    void clear() override;
     /// Feed a new hit to the tracks recognition algorithm
-    void addHit( const CTPPSDiamondRecHit& recHit );
-
+    void addHit( const CTPPSDiamondRecHit& recHit ) override;
     /// Produce a collection of tracks for the current station, given its hits collection
-    int produceTracks( edm::DetSet<CTPPSDiamondLocalTrack>& tracks );
+    int produceTracks( edm::DetSet<CTPPSDiamondLocalTrack>& tracks ) override;
 
   private:
-    struct HitParameters {
-      HitParameters( const float center, const float width ) :
-        center( center ), width( width ) {}
-      float center;
-      float width;
-    };
-    typedef std::vector<HitParameters> HitParametersVector;
-    typedef std::unordered_map<int,HitParametersVector> HitParametersVectorMap;
-
-    /// Default hit function accounting for the pad spatial efficiency
-    static const std::string pixelEfficiencyDefaultFunction_;
-
-    const float threshold_;
-    const float thresholdFromMaximum_;
-    const float resolution_;
-    const float sigma_;
-    const float startFromX_;
-    const float stopAtX_;
-
-    float yPosition_;
-    float yWidth_;
-    float yPositionInitial_;
-    float yWidthInitial_;
-    float zPosition_;
-    float zWidth_;
-    float zPositionInitial_;
-    float zWidthInitial_;
-
-    /// Function for pad efficiency
-    TF1 hit_f_;
-    HitParametersVectorMap hitParametersVectorMap_;
     std::unordered_map<int,int> mhMap_;
 };
 
+/****************************************************************************
+ * Implementation
+ ****************************************************************************/
+
+void
+CTPPSDiamondTrackRecognition::clear()
+{
+  CTPPSTimingTrackRecognition<CTPPSDiamondLocalTrack, CTPPSDiamondRecHit>::clear();
+  mhMap_.clear();
+}
+
+//----------------------------------------------------------------------------------------------------
+
+void
+CTPPSDiamondTrackRecognition::addHit( const CTPPSDiamondRecHit& recHit )
+{
+  // store hit parameters
+  hitVectorMap_[recHit.getOOTIndex()].emplace_back( recHit );
+}
+
+//----------------------------------------------------------------------------------------------------
+
+int
+CTPPSDiamondTrackRecognition::produceTracks( edm::DetSet<CTPPSDiamondLocalTrack>& tracks )
```

Comment:
remind me please why was this moved inline.

Reply:
Hi @slava77 
No strong reason as far as I can tell. Only a choice from one of the contributing authors. If you prefer we can move it back to the implementation file.

Reply:
please move to .cc

Reply:
:+1: